commercelayer/resource_manual_tax_calculator.go

```diff
@@ -0,0 +1,152 @@
+package commercelayer
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	commercelayer "github.com/incentro-dc/go-commercelayer-sdk/api"
+)
+
+func resourceManualTaxCalculator() *schema.Resource {
+	return &schema.Resource{
+		Description: "Configure the manual tax calculator by creating one or more associated tax rules. " +
+			"The rules will apply the related tax rate to the matching orders.",
+		ReadContext:   resourceManualTaxCalculatorReadFunc,
+		CreateContext: resourceManualTaxCalculatorCreateFunc,
+		UpdateContext: resourceManualTaxCalculatorUpdateFunc,
+		DeleteContext: resourceManualTaxCalculatorDeleteFunc,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description: "The manual payment unique identifier",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"type": {
+				Description: "The resource type",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"attributes": {
+				Description: "Resource attributes",
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				MinItems:    1,
+				Required:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Description: "The tax calculator's internal name.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"reference": {
+							Description: "A string that you can use to add any external identifier to the resource. This " +
+								"can be useful for integrating the resource to an external system, like an ERP, a " +
+								"marketing tool, a CRM, or whatever.",
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"reference_origin": {
+							Description: "Any identifier of the third party system that defines the reference code",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+						"metadata": {
+							Description: "Set of key-value pairs that you can attach to the resource. This can be useful " +
+								"for storing additional information about the resource in a structured format",
+							Type: schema.TypeMap,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+							Optional: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
```

Comment:
I am assuming the tax rules relation will follow in the follow-up pr about the tax rules?